pkg/webhookconfig/configmanager.go

```diff
@@ -338,8 +338,12 @@ func (m *webhookConfigManager) reconcileWebhook(namespace, name string) error {
 	logger := m.log.WithName("reconcileWebhook").WithValues("namespace", namespace, "policy", name)
 
 	policy, err := m.getPolicy(namespace, name)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return errors.Wrapf(err, "unable to get policy object %s/%s", namespace, name)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "unable to get policy object %s/%s", namespace, name)
+		} else {
+			policy = nil
```

Comment:
In this case we are deleting the policy now instead of just returning errors?

Reply:
Nope, the code below expects nil when policy is not found.
The client never returns nil though.

Reply:
Therefore setting it to nil here makes the remaining code valid.
(the issue was detected by golangcilint, took me a while to understand the root cause of the warning).